include/seqan3/search/dream_index/interleaved_bloom_filter.hpp

```diff
@@ -0,0 +1,518 @@
+     *
+     * \attention This function is only available for **uncompressed** Interleaved Bloom Filters.
+     * \attention The new number of bins must be greater or equal to the current number of bins.
+     *
+     * \details
+     *
+     * The resulting `seqan3::interleaved_bloom_filter` has an increased size proportional to the increase in the
+     * `bin_width` (the number of 64-bit words needed to represent `bins` many bins), e.g.
+     * resizing a `seqan3::interleaved_bloom_filter` with 40 bins to 73 bins also increases the `bin_width` from 1 to
+     * 2 and hence the new `seqan3::interleaved_bloom_filter` will be twice the size.
+     * This increase in size is necessary to avoid invalidating all computed hash functions.
+     * If you want to add more bins while keeping the size constant, you need to rebuild the
+     * `seqan3::interleaved_bloom_filter`.
+     *
+     * ### Example
+     *
+     * \include test/snippet/search/dream_index/interleaved_bloom_filter_increase_bin_number_to.cpp
+     */
+    void increase_bin_number_to(bin_count const new_bins)
+    //!\cond
+        requires data_layout_mode == data_layout::uncompressed
+    //!\endcond
+    {
+        if (new_bins.get() < bins_)
+            throw std::invalid_argument{"The value of new_bins must be >= the current number of bins."};
+
+        // Equivalent to ceil(new_bins.get() / 64)
+        size_t new_bin_width = (new_bins.get() + 63) >> 6;
+
+        bins_ = new_bins.get();
+        result_buffer.resize(bins_);
+
+        if (new_bin_width == bin_width) // No need for internal resize if bin_width does not change.
+            return;
+
+        size_t new_block_size = new_bin_width << 6;
+        size_t new_bits       = block_count * new_block_size;
+
+        size_t idx_{new_bits}, idx{data.size()};
+        size_t delta = new_block_size - block_size + 64;
+
+        data.resize(new_bits);
+
+        for (size_t i = idx_, j = idx; j > 0; i -= new_block_size, j -= block_size)
+        {
+            size_t stop = i - new_block_size;
+
+            for (size_t ii = i - delta, jj = j - 64; stop && ii >= stop; ii -= 64, jj -= 64)
+            {
+                uint64_t old = data.get_int(jj);
+                data.set_int(jj, 0);
+                data.set_int(ii, old);
+            }
+        }
+
+        bin_width = new_bin_width;
+        block_size = new_block_size;
+    }
+    //!\}
+
+    /*!\name Lookup
+     * \{
+     */
+    /*!\brief Determines set membership of a given value.
+     * \param[in] value The raw value to process.
+     *
+     * \attention The result of this function must always be bound via reference, e.g. `auto &`.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/search/dream_index/interleaved_bloom_filter_bulk_contains.cpp
+     */
+    [[nodiscard]] binning_bitvector const & bulk_contains(size_t const value) const & noexcept
```

Comment:
Since we deleted copy ctor/assignment for `binning_bitvector`, we cannot do `auto res = ibf. bulk_contains(23)`, which would copy the `binning_bitvector` into `res`.
We enforce `auto & ref = ...`